json-rpc/types/src/views.rs

```diff
@@ -859,3 +935,21 @@ impl TryFrom<AccountStateProof> for AccountStateProofView {
         })
     }
 }
+
+impl TryInto<AccountStateProof> for &AccountStateProofView {
+    type Error = Error;
+
+    fn try_into(self) -> Result<AccountStateProof, Self::Error> {
+        let ledger_info_to_transaction_info_proof: TransactionAccumulatorProof =
+            bcs::from_bytes(&self.ledger_info_to_transaction_info_proof.as_ref())?;
+        let transaction_info: TransactionInfo = bcs::from_bytes(&self.transaction_info.as_ref())?;
+        let transaction_info_with_proof =
+            TransactionInfoWithProof::new(ledger_info_to_transaction_info_proof, transaction_info);
+        let transaction_info_to_account_proof: SparseMerkleProof<AccountStateBlob> =
+            bcs::from_bytes(&self.transaction_info_to_account_proof.as_ref())?;
+        Ok(AccountStateProof::new(
+            transaction_info_with_proof,
+            transaction_info_to_account_proof,
+        ))
+    }
+}
```

Comment:
This works just fine, despite Orphan rules:
```
impl TryFrom<AccountStateProofView> for AccountStateProof {
    type Error = Error;

    fn try_from(value: AccountStateProofView) -> Result<Self, Self::Error> {
        let ledger_info_to_transaction_info_proof: TransactionAccumulatorProof =
            bcs::from_bytes(&value.ledger_info_to_transaction_info_proof.as_ref())?;
        let transaction_info: TransactionInfo = bcs::from_bytes(&value.transaction_info.as_ref())?;
        let transaction_info_with_proof =
            TransactionInfoWithProof::new(ledger_info_to_transaction_info_proof, transaction_info);
        let transaction_info_to_account_proof: SparseMerkleProof<AccountStateBlob> =
            bcs::from_bytes(&value.transaction_info_to_account_proof.as_ref())?;
        Ok(AccountStateProof::new(
            transaction_info_with_proof,
            transaction_info_to_account_proof,
        ))
    }
}
```

Reply:
oh huh, no kidding. changed!